From b1ad2e00c0277205d5a59bbf1442fbd717b68aea Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:54:29 -0400
Subject: [net] sunrpc/xprtrdma: Reset connection timeout after successful reconnect

Message-id: <1403798077-28799-71-git-send-email-steved@redhat.com>
Patchwork-id: 84928
O-Subject: [RHEL7.1 PATCH 70/78] xprtrdma: Reset connection timeout after successful reconnect
Bugzilla: 1113248
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Chuck Lever <chuck.lever@oracle.com>

If the new connection is able to make forward progress, reset the
re-establish timeout. Otherwise it keeps growing even if disconnect
events are rare.

The same behavior as TCP is adopted: reconnect immediately if the
transport instance has been able to make some forward progress.

Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
Signed-off-by: Anna Schumaker <Anna.Schumaker@Netapp.com>
Upstream Id: 18906972aa1103c07869c9b43860a52e0e27e8e5
Bugzilla: 1113248
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sunrpc/xprtrdma/rpc_rdma.c b/net/sunrpc/xprtrdma/rpc_rdma.c
index dc4a826..ac65b0c 100644
--- a/net/sunrpc/xprtrdma/rpc_rdma.c
+++ b/net/sunrpc/xprtrdma/rpc_rdma.c
@@ -770,6 +770,7 @@ repost:
 
  /* from here on, the reply is no longer an orphan */
  req->rl_reply = rep;
+ xprt->reestablish_timeout = 0;
 
  /* check for expected message types */
  /* The order of some of these tests is important. */
-- 
1.7.1